plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts

```diff
@@ -58,7 +58,10 @@ export class KubernetesFanOutHandler {
     requestBody: KubernetesRequestBody,
     objectTypesToFetch: Set<KubernetesObjectTypes> = DEFAULT_OBJECTS,
   ) {
-    const entityName = requestBody.entity.metadata.name;
+    const entityName =
+      requestBody.entity?.metadata?.annotations?.[
+        'backstage.io/kubernetes-label-selector'
+      ] || `backstage.io/kubernetes-id=${requestBody.entity?.metadata?.name}`;
```

Comment:
Looking a bit strange to me, I'm assuming we'd want to check if there's a `kubernetes-id` annotation and use that, and otherwise fall back to entity name? This seems to do something different

Reply:
oh so it should be this, right? : 
`const entityName =
      requestBody.entity?.metadata?.annotations?.[
        'backstage.io/kubernetes-id'
      ] || requestBody.entity?.metadata?.name;`

Reply:
Yep!

Reply:
Hi @Rugvip , thanks for the feedback, I have updated the code.